Homework/src/by/tms/lesson13/Runner.java

```diff
@@ -0,0 +1,180 @@
+package by.tms.lesson13;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import javax.xml.XMLConstants;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.TransformerFactoryConfigurationError;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.transform.stream.StreamSource;
+import javax.xml.validation.Schema;
+import javax.xml.validation.SchemaFactory;
+import javax.xml.validation.Validator;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Random;
+
+public class Runner {
+    /*
+    Создать xml-файл, хранящий информацию об объектах предметной области ниже. Для валидации полученного xml-файла необходимо разработать соответствующую xsd схему.
+1. Общие требования:
+- Простые и комплексные типы
+- Шаблоны и предельные значения
+- Обязательно использование атрибутов и типа ID
+- Задание атрибутов по умолчанию
+2. Произвести проверку XML-документа с помощью XSD.
+3. Предметная область:
+Компьютерные комплектующие:
+- Name - название комплектующего
+- Origin - страна производства
+- Price - цена (0 - n USD)
+- Type (должно быть несколько)
+- периферийное либо нет, энергопотребление (ватт), наличие кулера( есть или нет),
+ группа комплектующих (устройства ввода-вывода, мультимедийные), порты (COM, USB, LPT)
+- Critical - критично ли наличие комплектующего для работы компьютера.
+Корневой элемент назвать Devices
+     */
+    public static void main(String[] args) {
+        try {
+            DocumentBuilder documentBuilder = DocumentBuilderFactory.newInstance().newDocumentBuilder();
+            Document document = documentBuilder.parse("src/by/tms/lesson13/devices.xml");
+            addNewDevice(document);
+            Node root = document.getDocumentElement();
+            System.out.println("Devices:");
+            NodeList devices = root.getChildNodes();
+            for (int i = 0; i < devices.getLength(); i++) {
+                Node device = devices.item(i);
+                if (device.getNodeType() != Node.TEXT_NODE) {
+                    NodeList deviceProps = device.getChildNodes();
+                    for (int j = 0; j < deviceProps.getLength(); j++) {
```

Comment:
this cycle is pretty similar to parrent. Better to write one method with cycle and use it. 
Because if there will be structure like: 

`<devices>
    <device>
        <specification>
            <some_node>
                <some_more_node>
                </some_more_node>
            </some_node>
        </specification>
    </device>
</devices>`

you'll have much more internal cycles